src/app/borrower/borrower.component.spec.ts

```diff
@@ -1,25 +1,273 @@
+import {
+  async,
+  ComponentFixture,
+  TestBed,
+  fakeAsync,
+  tick,
+} from "@angular/core/testing";
+import { FormsModule, ReactiveFormsModule } from "@angular/forms";
+import { BorrowerComponent } from "./borrower.component";
+import { HLmsService } from "../common/h/hLms.service";
+import { Observable, from, of, observable, throwError } from "rxjs";
+import { CUSTOM_ELEMENTS_SCHEMA } from "@angular/core";
+import {
+  SortCopiesByBranchPipe,
+  SortCopiesByTitlePipe,
+} from "../common/h/sort-copies.pipe";
+import { HttpClientModule } from "@angular/common/http";
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from "@angular/common/http/testing";
 
-import { BorrowerComponent } from './borrower.component';
-
-describe('BorrowerComponent', () => {
+describe("BorrowerComponent", () => {
   let component: BorrowerComponent;
+  let service: HLmsService;
   let fixture: ComponentFixture<BorrowerComponent>;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ BorrowerComponent ]
-    })
-    .compileComponents();
+      declarations: [
+        BorrowerComponent,
+        SortCopiesByBranchPipe,
+        SortCopiesByTitlePipe,
+      ],
+      imports: [
+        ReactiveFormsModule,
+        HttpClientModule,
+        FormsModule,
+        HttpClientTestingModule,
+      ],
+      providers: [HLmsService],
+      schemas: [CUSTOM_ELEMENTS_SCHEMA],
+    }).compileComponents();
+    service = new HLmsService(null);
+    component = new BorrowerComponent(service);
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(BorrowerComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
   });
 
-  it('should create', () => {
+  it("should create", () => {
     expect(component).toBeTruthy();
   });
+
+  it("should get a borrower", fakeAsync(() => {
+    let mockBorrower = {
+      cardNo: 3,
+      name: "Esta Aldin",
+      address: "570 Rowland Circle",
+      phone: "906-381-0555",
+      loans: [
+        {
+          bookId: 14,
+          branchId: 1,
+          cardNo: 3,
+          dateOut: "2020-04-23T16:31:12.000Z",
+          dueDate: "2020-05-14T09:11:47.000Z",
+          dateIn: null,
+        },
+        {
+          bookId: 15,
+          branchId: 6,
+          cardNo: 3,
+          dateOut: "2020-04-27T16:38:08.000Z",
+          dueDate: "2020-05-14T15:54:56.000Z",
+          dateIn: null,
+        },
+        {
+          bookId: 17,
+          branchId: 6,
+          cardNo: 3,
+          dateOut: "2020-04-20T15:13:14.000Z",
+          dueDate: "2020-04-23T18:42:22.000Z",
+          dateIn: null,
+        },
+      ],
+    };
+
+    spyOn(service, "readBorrower").and.returnValue(
+      Promise.resolve(mockBorrower)
+    );
+    component.readBorrower(3);
+    tick();
+    expect(service).toBeTruthy();
+    expect(component.borrower).toEqual(mockBorrower);
+  }));
+
+  it("should get a borrower", fakeAsync(() => {
```

Comment:
Looks like you have this function in here twice.

Reply:
oops, good catch